### DISCUSSION TITLE
Sending request from another extension?

### DISCUSSION THREAD

**Discussion by theIDinside** [microsoft/vscode-cpptools#8341] [Ideas]
Is it possible to send a request, to cpptools from my own extension? I want it to send a "reverse-step" request, but there is absolutely no documentation on how to do it. in rr, we want to step backwards, and it uses gdb MI, so it works to type in ```-exec reverse-step``` in the debug console but that is horribly, just horribly unergonomic - so I'm trying to find ways to create an action that sends a request to cpptools to do that for me (since it's also defined in the protocol, if I'm not mistaken). 

If that's not possible, is there any possibility to copy & paste that into the debug console using some command?

**Comment by sean-mcmanus** [comment]
@WardenGnaw Do you know?

**Comment by WardenGnaw** [comment]
You will want to look at the `DebugAdapterTracker` APIs. 

This question is better suited for the [VS Code team](https://github.com/Microsoft/vscode) since that is an interface they provide extensions to call into debug adapters.
